docs/add-runtime.md

```diff
@@ -0,0 +1,250 @@
+
+Next set the permissions for the user you created so your build and run step will have access to them:
+```
+RUN ["chown", "-R", "openruntimes:openruntimes", "/usr/local/src"]
+RUN ["chown", "-R", "openruntimes:openruntimes", "/usr/code"]
+RUN ["chown", "-R", "openruntimes:openruntimes", "/usr/workspace"]
+RUN ["chown", "-R", "openruntimes:openruntimes", "/usr/builtCode"]
+```
+
+Finally, you'll add a `CMD` command. For an interpreted language this should be:
+```
+CMD ["/usr/local/src/start.sh"]
+```
+Since this will use your launch script when the runtime starts.
+
+For a compiled language this must be:
+```
+CMD ["tail", "-f", "/dev/null"]
+```
+so the build steps can be run.
+
+## 3. Building your Docker image and adding it to the list
+With your runtime successfully created you can now move on to building your Docker image and adding it to the script files used for generating all of the image files.
+
+Open up the `/runtimes/buildLocalOnly.sh` script first and add your runtime to it. The following is an example with dart version 2.12
+```
+echo 'Dart 2.12...'
+docker build -t openruntimes/dart:2.12 ./runtimes/dart-2.12
+```
+Next, open up the `/runtimes/build.sh` script and also add your runtime to it. This one is slightly different as this is the one that will be used for cross-platform compiles and deploying it to Docker hub. The following is an example also with dart version 2.12:
+```
+echo  'Dart 2.12...'
+docker buildx build --platform linux/amd64,linux/arm64 -t dart-runtime:2.12 ./runtimes/dart-2.12/ --push
+```
+
+## 4. Adding the runtime to the runtimes list
+In `src/Runtimes/Runtimes` create a new entry in the `__construct()` method in the Runtimes class like so:
+```
+$dart = new Runtime('dart', 'Dart');
+$dart->addVersion('2.12', 'dart-runtime:2.12', 'openruntimes-ubuntu:20.04', [System::X86, System::ARM]);
+$this->runtimes['dart'] = $dart;
+```
+This is an example of what you would do for a compiled language such as dart.
+
+The first line is creating a new language entry, The first parameter is the internal name and the second one is the external one which is what the user will see in projects that use OpenRuntimes such as Appwrite.
+
+The second line adds a new version to the language entry, I'll break down the parameters:
+```
+1: Version - The version of the runtime you are creating.
+2: Build Image - The image used to build the code
+3: Run Image - The image used to run the code. 
+For interpreted languages, this is normally the same as the Build Image, but for compiled languages, this can be either "openruntimes-alpine:3.13.6" or "openruntimes-ubuntu:20.04"
+We recommend using Alpine when possible and using Ubuntu if the runtime doesn't work on Alpine.
+4: Platforms Supported - These are the architectures this runtime is available to.
+```
+The third line simply adds the new runtime to the main list.
+
+## 5. Adding tests
+
+### 5.1 Writing your test execution script
```

Comment:
We should expalin what exactly a script in test needs to do.. It must be using library for HTTP client (to test library functionality), use some payload input formated as JSON, and return specific set of keys. We should explain that, and also point to tests in Base.php to see what expected input and output are.